assets/src/edit-story/panels/size.js

```diff
@@ -31,21 +33,61 @@ function SizePanel( { selectedElements, onSetProperties } ) {
 				{ 'Size' }
 			</Title>
 			<InputGroup
+				type="number"
 				label="Width"
 				value={ state.width }
 				isMultiple={ width === '' }
-				onChange={ ( value ) => setState( { ...state, width: isNaN( value ) || value === '' ? '' : parseFloat( value ) } ) }
+				onChange={ ( value ) => {
+					const newWidth = isNaN( value ) || value === '' ? '' : parseFloat( value );
+					setState( {
+						...state,
+						width: newWidth,
+						// @todo: Move to the reducer once available. This is especially critical
```

Comment:
Which reducer in particular? It's a pretty specific invariant.

Reply:
This is worth discussing. The business rules here are pretty specific and sensitive to what's actually specifically being updated. E.g. the width-only update may need to be reflected in the height for `keepRatio=true` elements. The `keepRatio` update may need to be updated in both width and height. And so on. This is all complicated in case if we update a width for multiple elements with different `keepRatio`/`origRatio` values. I think for us this means that:

1. We need to be able to run an update separately for each element.
2. We'd want to contain the update rules centrally.

Hence I'd like to see if we can move these rules to the reducer for element update dispatches that you're working on in the #3910.

Reply:
This requirement is removed per clarification with @samitron7. See this thread: https://github.com/ampproject/amp-wp/pull/3953/files#r359940148 for details.